state/settings.go

```diff
@@ -315,6 +315,27 @@ func (st *State) ReadSettings(collection, key string) (*Settings, error) {
 	return readSettings(st.db(), collection, key)
 }
 
+// readSettingsWithFallback attempts to retrieve settings first for the
+// requested key, then if not found, a non-empty fallback key.
+// If the fallback is used, the settings are created for the requested key.
+func readSettingsWithFallback(db Database, collection, requestKey, fallbackKey string) (*Settings, error) {
```

Comment:
The function name seems a bit misleading as the the function can potentially create the settings by copying the fallback ones.